### PR TITLE
Add docs for new ingest pipeline extension points in Fleet

### LINKED ISSUE #638: [REQUEST]: Document new ingest pipeline extension points in Fleet
### Description

In https://github.com/elastic/kibana/pull/170270 we extended the set of "extension points" for ingest pipelines that are included in Fleet's default component templates:

- `global`
- `${type}`
- `${type}-${package}`
- `${type}-${dataset}`

We should update https://www.elastic.co/guide/en/fleet/8.10/data-streams-pipeline-tutorial.html to mention these extension points for various levels of customization. Processors can be added globally, for all logs or all metrics, for an entire integration, or for a specific dataset. 

### Resources

- https://github.com/elastic/kibana/pull/170270
- https://www.elastic.co/guide/en/fleet/8.10/data-streams-pipeline-tutorial.html

### Collaboration

The documentation team will investigate the issue and create the initial content.

### Point of contact.

**Main contact:** @nchaulet + @kpollich 

**Stakeholders:**: @elastic/fleet

### PR DESCRIPTION
**Pull request by kilfoyle**
This adds docs for the new ingest pipeline extension points:
 - Updates the "[Ingest pipelines](https://ingest-docs_675.docs-preview.app.elstc.co/guide/en/fleet/master/data-streams.html#data-streams-pipelines)" section with a description and example of each level of customization.
 - Adds a new "[Step 6](https://ingest-docs_675.docs-preview.app.elstc.co/guide/en/fleet/master/data-streams-pipeline-tutorial.html#data-streams-pipeline-six)" to the ingest pipelines tutorial, showing how to process all log events.

Closes #638
Target: 8.12.0

@nchaulet and @kpollich I hope I've understood these extension points correctly. :-)

### CODE REVIEW

File: docs/en/ingest-management/data-streams.asciidoc [hunk 1]
```diff
@@ -122,10 +122,94 @@ Starting in version 8.4, all default ingest pipelines call a non-existent and no
 If left uncreated, this pipeline has no effect on your data. However, if added to a data stream and customized,
 this pipeline can be used for custom data processing, adding fields, sanitizing data, and more.
 
-The full name of the `@custom` pipeline follows the following pattern: `<type>-<dataset>@custom`.
-The `@custom` pipeline can directly contain processors or you can use the
-pipeline processor to call other pipelines that can be shared across multiple data streams or integrations.
-The `@custom` pipeline will persist across all version upgrades.
+Staring in version 8.12, ingest pipelines can be configured to process events at various levels of customization.
+
+`@global`::
```
Comment (nchaulet) [inline docs/en/ingest-management/data-streams.asciidoc:127]: The name of the global custom pipeline is `global@custom`

Comment (kilfoyle) [reply docs/en/ingest-management/data-streams.asciidoc]: This is fixed by [d8bf6e1](https://github.com/elastic/ingest-docs/pull/675/commits/d8bf6e10217ffe3ccb54f265515d3b59fa61bd12)

**Comment by kilfoyle** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by kilfoyle** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Review by nchaulet — APPROVED** [review]
This looks good to me 🚀

**Review by jrodewig — APPROVED** [review]

**Pull request merged**